### PR TITLE
gui-libs/wlroots: fix automagic of libliftoff and lcms, drop USE=tinywl; add gui-wm/tinywl

### PR DESCRIPTION
**Pull request by sevz17**
---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: gui-wm/tinywl/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person" proxied="yes">
```
Comment (sevz17) [inline gui-wm/tinywl/metadata.xml:4]: @arthurzam @prometheanfire do you want me to add you as maintainers?

**Comment by thesamesam** [comment]
Please update the commit messages to explain rationale.

File: gui-wm/tinywl/tinywl-0.17.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+LICENSE="CC0-1.0"
+SLOT="0"
+DEPEND="
+	dev-libs/wayland
+	x11-libs/libxkbcommon
+	=gui-libs/wlroots-$(ver_cut 1-2)*:=
+"
+RDEPEND="
+	${DEPEND}
+	!!gui-libs/wlroots[tinywl(-)]
```
Comment (thesamesam) [inline gui-wm/tinywl/tinywl-0.17.3.ebuild:27]: A hard blocker isn't right here and it prevents the dependency being resolved cleanly by Portage (it requires manual intervention). Hard blockers are only needed if something can't even be built if the other thing is installed, as opposed to a file collision which "allows" a file collision.

File: gui-wm/tinywl/tinywl-0.17.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	virtual/pkgconfig
+"
+
+src_prepare() {
+	sed -i -e "s/-Werror//" tinywl/Makefile || die
+	default
+}
+
+src_compile() {
+	make -C tinywl
```
Comment (thesamesam) [inline gui-wm/tinywl/tinywl-0.17.3.ebuild:41]: This should be `emake` to handle failing and also parallel jobs.

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: gui-libs/wlroots/files/wlroots-0.17-fix-automagic-libliftoff.patch
Comment (ezzieyguywuf) [inline gui-libs/wlroots/files/wlroots-0.17-fix-automagic-libliftoff.patch:1]: Tested this patch locally, it does solve the build-time error with `USE=-liftoff`

**Review by arthurzam — APPROVED** [review]

**Pull request closed without merging**